### PR TITLE
Add deployment scripts docs

### PR DESCRIPTION
**Pull request by THenry14**
Closes https://github.com/foundry-rs/starknet-foundry/issues/1135

## Introduced changes

- add deployment scripts docs

## Checklist

- [X] Linked relevant issue
- [X] Updated relevant documentation
- [X] Added relevant tests
- [X] Performed self-review of the code
- [X] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: docs/src/starknet/script.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+# Cairo Deployment Scripts
```
Comment (abulenok) [inline docs/src/starknet/script.md:1]: Let's add something about using `path_to_scarb_toml` when running `sncast script` from arbitrary directory

Comment (THenry14) [reply docs/src/starknet/script.md:1]: We mention this in several places in the docs, but never in sections dedicated to specific subcommands. Do you think we should change that? I guess this is rather top-level stuff

File: docs/src/starknet/script.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+[dependencies]
+starknet = ">=2.3.0"
+sncast_std = { git = "https://github.com/foundry-rs/starknet-foundry.git", tag = "v0.11.0" }
+```
+
+To run the script, do:
+
+```shell
+$ sncast \
+  --rpc-url http://127.0.0.1:5050 \
```
Comment (abulenok) [inline docs/src/starknet/script.md:76]:
```suggestion
  --url http://127.0.0.1:5050 \
```
https://foundry-rs.github.io/starknet-foundry/appendix/cast/common.html#--url--u-rpc_url

Comment (THenry14) [reply docs/src/starknet/script.md]: great catch! chnged in newest commit

File: docs/src/starknet/script.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+```
+
+Please note that `map` contract was specified as the dependency. In our example, it resides in the filesystem. It has to
+be compiled in `build-external-contracts`.
+
+To run the script, do:
+
+```shell
+$ sncast \
+  --rpc-url http://127.0.0.1:5050 \
```
[the remark this answers is not in the document]
Comment (THenry14) [reply docs/src/starknet/script.md]: done :+1:

File: docs/src/starknet/script.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+[[target.starknet-contract]]
+sierra = true
+casm = true
+build-external-contracts = [
+    "map::Map"
+]
+```
+
+Please note that `map` contract was specified as the dependency. In our example, it resides in the filesystem. It has to
+be compiled in `build-external-contracts`.
```
Comment (abulenok) [inline docs/src/starknet/script.md:159]:
```suggestion
Please note that `map` contract was specified as the dependency. In our example, it resides in the filesystem. To generate the artifacts for it that will be accessible from the script you need to use the `build-external-contracts` property.
```

File: docs/src/starknet/script.md (hunk 1 above)
Comment (abulenok) [reply docs/src/starknet/script.md:1]: okay, it can stay as is then

**Review by abulenok — APPROVED** [review]

**Pull request merged**